### PR TITLE
[XeTLA] Sync 0.3.7.1

### PR DESCRIPTION
**Pull request by sunjiweiswift**
## Type of Change

feature or bug fix or documentation or others: other code sync
API changed or not: not

## Description
Dg2 A770M
============== [kernel time] gflops   ================== 
[kernel time]The minimum gflops(GPU_time) is 494.4
[kernel time]The maximum gflops(GPU_time) is 857.426
[kernel time]The median  gflops(GPU_time) is 830.218
[kernel time]The mean    gflops(GPU_time) is 812.764
======================================================
Xe 

***************** PROFILING FOR KERNEL0 ***********************
============= Profiling for [kernel time] =============
[kernel time]The first   running(GPU_time) time is 0.12544ms
[kernel time]The minimum running(GPU_time) time is 0.06384ms
[kernel time]The maximum running(GPU_time) time is 0.12544ms
[kernel time]The median  running(GPU_time) time is 0.0664ms
[kernel time]The mean(exclude the first trial) running(GPU_time) time is 0.0672145ms
[kernel time]The variance(exclude the first trial) running(GPU_time) time is 4.38324e-05ms
======================================================
============== [kernel time] gflops   ==================
[kernel time]The minimum gflops(GPU_time) is 802.58
[kernel time]The maximum gflops(GPU_time) is 1577
[kernel time]The median  gflops(GPU_time) is 1516.2
[kernel time]The mean    gflops(GPU_time) is 1497.82
======================================================

detail description 
Issues: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: include/common/utils/raw_send_nbarrier.hpp [hunk 1]
```diff
@@ -102,8 +102,7 @@ struct xetla_nbarrier_t<num_producers, num_consumers, gpu_arch::Dg2> {
     /// note:  all subgroups participating the barrier should have the same
     /// barrier_id. Here is the bspec link
     /// https://gfxspecs.intel.com/Predator/Home/Index/54006
-    __XETLA_API void init_nbarrier(uint8_t nbarrier_id,
-            nbarrier_role role = nbarrier_role::producer_consumer) {}
+    __XETLA_API void init_nbarrier(uint8_t, nbarrier_role) {}
```
Comment (airMeng) [inline include/common/utils/raw_send_nbarrier.hpp:105]: confirmed?

Comment (sunjiweiswift) [reply include/common/utils/raw_send_nbarrier.hpp:105]: yes splict_barrier don't init

File: examples/08_scaled_dot_product_attention/scaled_dot_product_attention.cpp [hunk 2]
```diff
@@ -424,9 +419,6 @@ void sdp_fwd_run(uint32_t iter) {
                     uint32_t boundary_n = (start_n + wg_tile_n_sv) > matrix_n
                             ? matrix_n
                             : (start_n + wg_tile_n_sv);
-                    uint32_t boundary_m = (start_m + wg_tile_m_sv) > matrix_m
-                            ? matrix_m
-                            : (start_m + wg_tile_m_sv);
```
[the remark this answers is not in the document]
Comment (sunjiweiswift) [reply examples/08_scaled_dot_product_attention/scaled_dot_product_attention.cpp:429]: Here is the modification of xetla/main

**Review by airMeng — APPROVED** [review]
please verify on arc and pvc

**Comment by sunjiweiswift** [comment]
> please verify on arc and pvc

done

**Review by zhewang1-intc — APPROVED** [review]

**Pull request merged**